docs/learning/ECSQLNullBehaviors.md

```diff
@@ -27,7 +28,11 @@ Attempting to update any of the properties (`intProp`, `arrBoolProp`, `structPro
 
 ## Setting some complex property children to null
 
-To be added
+This section would cover partially setting a complex property (point2d, struct, array) to null.
+
+- A `point2d` type property must always have all its coordinates present, hence updating a point2d to be partially null is not a valid input and would result in the property not being updated. This is why updating `p2dProp` to `{ x: null, y: 2.5 }` would be invalid and take no effect on the element's property.
+- A `struct` type property can have all of their children properties set to null. The child property would end up being cleared from the parent object with the remaining properties remaining intact. If all of the struct's children become null, the struct itself is considered to be null. This means that if `structProp` would include `{ doubleProp: 41.0, stringProp: null }`, the string property would be cleared with the doubleProp successfully updated to the expected value.
+- An `array` type (primitive or struct) property can store null values as any other value of the respective type. It would happen to be represented alongside the ordinary properties as null. However, similarly to structs, if all of the array's entries are nulls, the array itself is considered to be null. E.g. updating `arrBoolProp` to `[false, true, null, false]` is a perfectly valid data entry for a boolean type array property.
```

Comment:
We should agree for whether an array [null, null, null] is considered null itself as well or not.